addons/mail/static/src/core/common/quick_reaction_menu.js

```diff
@@ -34,7 +34,17 @@ export class QuickReactionMenu extends Component {
                 popoverClass: "o-mail-QuickReactionMenu-pickerPopover",
             }
         );
-        this.dropdown = useState(useDropdownState());
+        this.dropdown = useState(
+            useDropdownState({
+                onClose: () => {
+                    if (this.props.message.thread.messageInEdition) {
+                        this.props.message.thread.messageInEdition.composer.autofocus++;
+                    } else {
+                        this.props.message.thread.composer.autofocus++;
+                    }
```

Comment:
Also this when done from a mailbox would probably not take the correct thread into account.

We don't want the message origin thread, but the currently viewed thread.